### PR TITLE
[UNIT-TEST] MessageRepo sync messages test added

### PR DESCRIPTION
**Pull request by sidsharma2002**

### CODE REVIEW

**Comment by sidsharma2002** [comment]
@EngrTaofeek can you review this??

File: app/src/test/java/org/greenstand/android/TreeTracker/Extension.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package org.greenstand.android.TreeTracker
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+fun <T> LiveData<T>.getOrAwaitValue(
```
Comment (Elforama) [inline app/src/test/java/org/greenstand/android/TreeTracker/Extension.kt:9]: It doesn't look like anything in this file is used. Can we remove these methods?

File: app/src/test/java/org/greenstand/android/TreeTracker/models/messages/MessagesRepoTest.kt [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            it.toString(),
+            false,
+            it % 2 == 0
+        )
+    }
+
+    private val lastTimeMessageSynced = "123456789"
+
+    @Test
+    fun syncMessagesTest() = mainCoroutineRule.runBlockingTest {
```
Comment (Elforama) [inline app/src/test/java/org/greenstand/android/TreeTracker/models/messages/MessagesRepoTest.kt:89]: This method test name is vague, can we give it more detail?

**Review by Elforama — CHANGES_REQUESTED** [review]
Looks great, just left some minor comments.

File: app/src/test/java/org/greenstand/android/TreeTracker/models/messages/MessagesRepoTest.kt (hunk 2 above)
Comment (sidsharma2002) [reply app/src/test/java/org/greenstand/android/TreeTracker/models/messages/MessagesRepoTest.kt:89]: @Elforama  I have none coming in my mind, can you suggest one?

File: app/src/test/java/org/greenstand/android/TreeTracker/Extension.kt (hunk 1 above)
Comment (sidsharma2002) [reply app/src/test/java/org/greenstand/android/TreeTracker/Extension.kt:9]: ok

**Comment by Elforama** [comment]
Rebase this and I will merge it.

**Comment by sidsharma2002** [comment]
> Rebase this and I will merge it.

done

**Review by Elforama — APPROVED** [review]

**Pull request merged**